### PR TITLE
Cgroup V2 statistic engine improvements

### PR DESCRIPTION
**Pull request by gkhandake**
- Improve statistic engine to support cgroup v2 memory stats

### CODE REVIEW

File: tools/common-code/cmonitor_statistics_engine.py [hunk 1]
```diff
@@ -102,16 +103,21 @@ def dump_cpu_stats(self, verbose) -> dict:
     def dump_cpu_throttle_stats(self, verbose) -> dict:
         return self.cpu_throttle.dump_json(verbose)
 
-    def insert_memory_stats(self, stats: dict, sample_index: int) -> None:
-        if "stat.rss" in stats:
-            self.memory.insert_stat(stats["stat.rss"])
+    def insert_memory_stats(self, stats: dict, sample_index: int, cgroup_version: int) -> None:
+        stat_label = "stat.rss" if cgroup_version == 1 else "stat.file"
```
Comment (f18m) [inline tools/common-code/cmonitor_statistics_engine.py:107]: when we are dealing with cgroup_version==2, I guess we should be using the "file" + "anon" metric, if we want to have something similar to the cgroupv1 "rss" metric... what do you think?

**Comment by f18m** [comment]
@gkhandake can you merge from main? 
I was not sure about the changes to the Dockerfile.builder, so I had a look and I found my own docker pattern to be quite weird (with the dockerfile being split in a builder and non-builder part... for such a small app). So I corrected the thing and restored a more standard approach to docker image building.
If you merge from main, you can see if you can build the docker image without the need to change the Alpine pkgs being installed... thanks

**Comment by gkhandake** [comment]
> @gkhandake can you merge from main? I was not sure about the changes to the Dockerfile.builder, so I had a look and I found my own docker pattern to be quite weird (with the dockerfile being split in a builder and non-builder part... for such a small app). So I corrected the thing and restored a more standard approach to docker image building. If you merge from main, you can see if you can build the docker image without the need to change the Alpine pkgs being installed... thanks

**Comment by gkhandake** [comment]
Hi @f18m can we please also get a release with minor version bump ?

**Comment by f18m** [comment]
> Hi @f18m can we please also get a release with minor version bump ?

Sure, let me spin a new release.

**Pull request merged**